src/lpython/semantics/python_intrinsic_eval.h

```diff
@@ -77,20 +77,34 @@ struct IntrinsicNodeHandler {
                 char *c = ASR::down_cast<ASR::StringConstant_t>(
                                     ASRUtils::expr_value(arg))->m_s;
                 int ival = 0;
+                std::string str;
                 char *ch = c;
-                if (*ch == '-') {
+                if (*ch == '-' || *ch == '+') {
                     ch++;
                 }
                 while (*ch) {
+                    if(*ch == ' '){
+                        ch++;
+                        continue;
+                    }
+                    if(*ch == '\\'){
+                        ch++;
+                        if (*ch == 'n' || *ch == 'r' || *ch == 't') {
+                            ch++;
+                            continue;
+                        }
+                        throw SemanticError("invalid literal for int() with base 10: '"+ std::string(c) + "'", arg->base.loc);
+                    }
                     if (*ch == '.') {
                         throw SemanticError("invalid literal for int() with base 10: '"+ std::string(c) + "'", arg->base.loc);
                     }
                     if (*ch < '0' || *ch > '9') {
                         throw SemanticError("invalid literal for int() with base 10: '"+ std::string(c) + "'", arg->base.loc);
                     }
```

Comment:
I think the above could be simplified as follows:
```suggestion
                    if(*ch == ' ' || *ch == '\n' || *ch == '\r' || *ch == '\t'){
                        ch++;
                        continue;
                    }
                    if (*ch < '0' || *ch > '9') {
                        throw SemanticError("invalid literal for int() with base 10: '"+ std::string(c) + "'", arg->base.loc);
                    }
```

Reply:
I think `ch` holds the whole escaped character. So, we need not check for `\n` (or others) in two steps. The above approach performs the check in one step.

Example:
```cpp
#include <iostream>
#include <cstring>
using namespace std;

int main()
{
    char *mystr = "abc def \n\r\t pqr xyz";
    char *ch = mystr;
    while (*ch) {
        if (*ch == ' ') {
            // do not print anything
        } else if (*ch == '\n') {
            std::cout << "newline character" << std::endl;
        } else if (*ch == '\r') {
            std::cout << "carriage return" << std::endl;
        } else if (*ch == '\t') {
            std::cout << "tab character" << std::endl;
        } else {
            std::cout << *ch << std::endl;
        }
        ch++;
    }
    return 0;
}
```
Output:
```bash
a
b
c
d
e
f
newline character
carriage return
tab character
p
q
r
x
y
z
```